File: code/java/Enum.java
public class Enum {
// BEGIN_CODE
	enum BasicEnum {
		Option1,
		Option2
	}

	enum CharEnum {
		OptionA('a'),
		OptionB('b');

		private char v;

		CharEnum(char v) {
			this.v = v;
		}
	}

	public static void main(String[] args) {
		BasicEnum e1 = BasicEnum.Option1;
		CharEnum e2 = CharEnum.OptionA;

		if(e1.equals(BasicEnum.Option2))
			e2 = CharEnum.OptionB;

		if(e2.equals(CharEnum.OptionB))
			e1 = BasicEnum.Option2;
	}
// END_CODE
}

File: code/rust/enum.rs
#[allow(unused_variables)]
#[allow(unused_assignments)]
#[allow(dead_code)]
fn main() {
// BEGIN_CODE
    #[derive(PartialEq, Eq)]
    enum BasicEnum {
        Option1 = 1,
        Option2
    }

    enum CharEnum {
        OptionA(char),
        OptionB(char)
    }

    let mut e1 = BasicEnum::Option1;
    let mut e2 = CharEnum::OptionA('a');

    if e1 == BasicEnum::Option2 {
        e2 = CharEnum::OptionB('b');
    }

    match e2 {
        CharEnum::OptionB(..) => e1 = BasicEnum::Option2,
        _ => {}
    }
// END_CODE
}
